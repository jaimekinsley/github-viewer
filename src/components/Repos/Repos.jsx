import React from 'react';
import PropTypes from 'prop-types';

const Repos = ({ repos }) => {
  const repoElements = repos.map(repo => (
    <li key={`${repo.url}+${repo.name}`}>
      <a href={repo.url}> {repo.name} </a>
    </li>
  ));

  return (
    <ul>
      {repoElements}
    </ul>
  );
};

Repos.propTypes = {
  repos: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Repos;
