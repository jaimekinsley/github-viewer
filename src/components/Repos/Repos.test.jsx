import React from 'react';
import { shallow } from 'enzyme';
import Repos from './Repos';

describe('Repos component', () => {
  it('renders Repos', () => {
    const wrapper = shallow(<Repos repos={[
      { name:'about-me', url:'http://github.com/about-me' },
      { name:'bus-mall', url:'http://github.com/bus-mall' }
    ]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
