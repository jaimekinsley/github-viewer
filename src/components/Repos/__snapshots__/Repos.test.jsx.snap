// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component renders Repos 1`] = `
<ul>
  <li
    key="http://github.com/about-me"
  >
    <p>
      about-me
    </p>
    <a
      href="http://github.com/about-me"
    />
  </li>
  <li
    key="http://github.com/bus-mall"
  >
    <p>
      bus-mall
    </p>
    <a
      href="http://github.com/bus-mall"
    />
  </li>
</ul>
`;
